src/scraper/scraper.py

```diff
@@ -0,0 +1,74 @@
+"""
+    Main module interface
+"""
+from typing import List, Type
+
+import scrapy
+
+from scraper.scrapers.base_scraper import BaseScraper
+from .settings import URL_TO_SCRAPER
+
+
+def scrape(url: str) -> dict:
+    """
+        Scrape data for the given url if such url is scrappable,
+        Raise ValueError if not. 
+
+        Params:
+            + url - str : Url to scrape
+        Return:
+            A dict object, each describing the data that could be 
+            extracted for this url. Obtained data depends on the url itself, 
+            so available data may change depending on the scrapped url.
+            Dict format:
+             {
+                 "url" : (str) url where the data came from,
+                 "data": (dict) Data scraped for this url
+             }
+    """
+    scraper = _get_scraper_from_url(url)()
+    return scraper.scrape(url)
+
+
+def bulk_scrape(urls: List[str]) -> List[dict]:
+    """
+        Performs a bulk scraping over a list of urls.
+        Order in the item list it's not guaranteed to be
+        the same as in the input list
+
+        Parameters:
+            + urls : [str] = Urls to be scraped
+        Return:
+            A list of items scraped for each url in the original list
+    """
+
+    items = []
+    scrapers = {}
+    for url in urls:
+        # Classify urls to its according scraper
+        scraper = _get_scraper_from_url(url)
+
+        if not (url_list := scrapers.get(scraper)):
+            url_list = scrapers[scraper] = []
+
+        url_list.append(url)
+
+    # Bulk scrape urls
+    for (scraper, url_list) in scrapers.items():
+        s = scraper()  # Create a new scraper instance
+        items.extend(s.bulk_scrape(url_list))
+
+    return items
+
+
+def _get_scraper_from_url(url: str) -> Type[BaseScraper]:
+    """
+        Validates if this url is scrapable and returns its 
+        corresponding spider when it is
+    """
+
+    for known_url in URL_TO_SCRAPER.keys():
+        if known_url in url:
+            return URL_TO_SCRAPER[known_url]
```

Comment:
here for efficiency I would suggest: 

- validating that the URL is valid 
- extracting the domain name from the URL 
- using the domain name as the lookup key in the map

Reply:
In order to validate the URL, the two easier options I found are:
* Using a regex like [this](https://github.com/django/django/blob/stable/1.3.x/django/core/validators.py#L45) from django
* Using a validators library like [this](https://validators.readthedocs.io/en/latest/)

What approach do you think is the best? I would prefer to the first one since it does not requires a new dependency